# HG changeset patch
# User Rik <rik@octave.org>
# Date 1513478501 28800
#      Sat Dec 16 18:41:41 2017 -0800
# Node ID fc63541148647781f57398a3690622dd7df5db9b
# Parent  e1390a99a9931237ef42dbb40ef21cf179f8fcdf
# Parent  13344f00c564e2a76d903ec5097c08c363d1aeca
maint: merge stable to default.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -1727,17 +1727,17 @@ children_property::do_get_children (bool
 
 void
 children_property::do_delete_children (bool clear)
 {
   for (auto& hchild : children_list)
     {
       graphics_object go = gh_manager::get_object (hchild);
 
-      if (go.valid_object ())
+      if (go.valid_object () && ! go.get_properties ().is_beingdeleted ())
         gh_manager::free (hchild);
     }
 
   if (clear)
     children_list.clear ();
 }
 
 bool
diff --git a/scripts/io/dlmwrite.m b/scripts/io/dlmwrite.m
--- a/scripts/io/dlmwrite.m
+++ b/scripts/io/dlmwrite.m
@@ -155,17 +155,17 @@ function dlmwrite (file, M, varargin)
 
   ## Expand '\t' to TAB for Matlab compatibility
   if (strcmp (delim, '\t'))
     delim = "\t";
   endif
 
   if (ischar (file))
     [fid, msg] = fopen (file, opentype);
-  elseif (isscalar (file) && isnumeric (file))
+  elseif (is_valid_file_id (file))
     [fid, msg] = deal (file, "invalid file number");
   else
     error ("dlmwrite: FILE must be a filename string or numeric FID");
   endif
 
   if (fid < 0)
     error (["dlmwrite: " msg]);
   else
@@ -189,17 +189,17 @@ function dlmwrite (file, M, varargin)
       M = M.';
       b = zeros (2*rows (M), columns (M));
       b(1: 2 : end, :) = real (M);
       b(2: 2 : end, :) = imag (M);
       fprintf (fid, template, b);
     else
       fprintf (fid, template, M.');
     endif
-    if (! isscalar (file))
+    if (ischar (file))
       fclose (fid);
     endif
   endif
 
 endfunction
 
 
 %!test
